woocommerce/woocommerce-ios
# Appstore screenshot: Simulate in-person payment dialog for screenshot

⚠️ Please make sure that #7216 is reviewed and merged first: ⚠️  

### Description

This PR continues the work from #7216 to capture the collect payment dialog for app store screenshot. Changes include:

- Hardware: we need to create a mock card reader and card reader status, so public initializers for these had to be added.
- UI Layer: additional accessibility identifiers to access the buttons and detect the collect payment modal.
- Woo Foundation: a minor update to `CurrencyFormatter` was added to transform value from non-Latin to Latin. This is necessary to format the order total value from Arabic numerals to Latin ones. `CollectOrderPaymentUseCase` needs to check this value for collecting payments.
- Yosmite: add mocks for `OrderCardPresentPaymentEligibilityAction`, `SystemStatusAction` and `CardPresentPaymentAction`.

### Testing instructions

- Turn off the feature flag `.splitViewInOrdersTab` in `DefaultFeatureFlagService`.
- Optionally set the simulator to a non-English language to make sure the test isn’t relying on anything English-specific. Try Arabic to check the case with Arabic numerals.
- Select target WooCommerceScreenshots. Run the `testScreenshots()` UI test.
- Confirm that the test succeeds.

You can find the screenshots in the path `~/Library/Caches/tools.fastlane/screenshots`. It's likely that the folder `screenshots` hasn't been created yet, so you might need to create one yourself.

💡 A tip for changing the test language to Arabic: update the language in `~/Library/Caches/tools.fastlane/language.txt` to `ar`.

---
